src/Four_Player_Adapter.md

```diff
@@ -0,0 +1,197 @@
+network. The proper response is to send 0x88 *after* receiving the ID Byte and
+STAT1, in which case the upper-half of STAT1, STAT2, and STAT3 are updated to
+show that a Game Boy is "connected". If for whatever reason, the
+acknowledgement codes are not sent, the above bits are unset.
+
+Some examples of ping packets are shown below:
+
+Packet                | Description
+----------------------|-------------------------------------------------------
+`0xFE 0x01 0x01 0x01` | Ping packet received by Player 1 with no other Game Boys connected.
+`0xFE 0x11 0x11 0x11` | Ping packet received by Player 1 when Player 1 has connected.
+`0xFE 0x31 0x31 0x31` | Ping packet received by Player 1 when Players 1 & 2 have connected.
+`0xFE 0x71 0x71 0x71` | Ping packet received by Player 1 when Players 1, 2, & 3 have connected.
+`0xFE 0x62 0x62 0x62` | Ping packet received by Player 2 when Players 2 & 3 are connected (but not Player 1).
+
+It's possible to have situations where some players are connected but others
+are not; the gaps don't matter. For example, Player 1 and Player 4 can be
+connected, while Player 2 and Player 3 can be disconnected (or non-existant,
+same thing); most games do not care so long as Player 1 is active, as that
+Game Boy acts as master and orchestrates the multiplayer session from a
+software point of view. Because of the way the DMG-07 hardcodes player IDs
+based on which port a Game Boy is physically connected to, in the above
+situation Player 4 wouldn't suddenly become Player 2.
+
+During the ping phase, the master Game Boy is capable of setting up two
+parameters that will be used during the transmission phase. The clock rate for
+the transmission phase can be adjusted, as well as the packet size each Game
+Boy will use. The master Game Boy needs to respond with one byte for STAT2
+and STAT3 respectively. The chart below illustrates how a master Game Boy
+should respond to all bytes in a ping packet:
+
+```
+----------------------------
+DMG-07		Game Boy
+----------------------------
+0xFE	<-->	(ACK1) = 0x88
+STAT1	<-->	(ACK2) = 0x88	
+STAT2	<-->	(RATE) = Link Cable Speed 
+STAT3	<-->	(SIZE) = Packet Size
+```
+
+The new clock rate is only applied when entering the transmission phase; the
+ping phase runs at a constant 2048 bits-per-second. The formula for the ne
+clock rate is as follows:
+
+```
+DMG-07 Bits-Per-Second --> 4194304 / ((6 * RATE) + 512)
+```
+
+The lowest setting (RATE = 0) runs the DMG-07 at the normal speed DMGs usually
+transfer data (1KB/s), while setting it to 0xFF runs it close to the slowest
+speed (2042 bits-per-second).
+
+SIZE sets the length of packets exchanged between all Game Boys. Nothing fancy,
+just the number of bytes in each packet. It probably shouldn't be set to zero.
+
+### Transmission Phase
+
+When the master Game Boy (Player 1) is ready, it should send 4 bytes
+(`0xAA 0xAA 0xAA 0xAA`). (TODO: Are 4 bytes actually required?) This alerts
```

Comment:
I'm not sure TODOs should be scattered around like this

Reply:
Fair; what would you propose instead?